03-task-manager/starter/QuizAnswers2.txt

```diff
@@ -0,0 +1,20 @@
+1.In this lesson, you created a middleware function called asyncWrapper. Why?
+
+We created a middleware function called asyncWrapper to avoid setting up try and catch blocks for every function-controller (getAllTasks, createTask...); so we wanted to refactor our code. 
+In asyncWrapper we set up try catch blocks.
+asyncWrapper returns another async function that has access to req, res, next (from express.js).
+In the function body of current function-controller (getAllTasks, createTask...) we decide what to do with our argument: for example, in getAllTasks - we are waiting (await) to find all our tasks ( .find({}))
+and if everything is OK we send back all tasks to the user :  res.status(200).json({ tasks })
```

Comment:
Yes exactly! If you want to read more about error handling in Express, check out this post: https://thecodebarbarian.com/80-20-guide-to-express-error-handling